platform/platform-tests/testSrc/com/intellij/openapi/progress/impl/ProgressIndicatorTest.java

```diff
@@ -856,4 +856,23 @@ private static void checkIndicatorNotThrowInThisOldStaleDisgustingNonCancelableS
     catch (ProcessCanceledException ignored) {
     }
   }
+
+  public void testProgressIndicatorNotOverriddenIn_InRunProcessWithProgressSynchronously_duringWriteAction() {
+    ProgressIndicator progressIndicator = new ProgressIndicatorStub() {
+      @Override
+      public void processFinish() {
```

Comment:
What's this processFinish for?

Reply:
Need to override it otherwise com/intellij/openapi/progress/impl/CoreProgressManager.java:169 throws a runtime exception.

Reply:
You could use a more normal indicator like ProgressIndicatorBase instead of this test-only one.